### PR TITLE
Add Divisions collection, Add bulkDeleteBroadcastEmails, divisionSearchMembers, and divisionSearchTeams methods

### PR DESCRIPTION
**Pull request by dmccraw**

### CODE REVIEW

File: docs/collections/broadcast_emails.md [hunk 1]
```diff
@@ -115,3 +116,25 @@ teamsnap.saveBroadcastEmail(broadcastEmail).then(function(){
...
 });
 ```
+
+
+---
+
+
+<a id="bulkDeleteBroadcastEmails"></a>
+## `bulkDeleteBroadcastEmails(broadcastEmailIds, callback)`
+Bulk deletes a list of `broadcastEmail` item.
```
Comment (matthopson) [inline docs/collections/broadcast_emails.md]: "items"

File: docs/collections/broadcast_emails.md [hunk 2]
```diff
@@ -115,3 +116,25 @@ teamsnap.saveBroadcastEmail(broadcastEmail).then(function(){
...
+
+### Params
+* `broadcastEmailIds`: [array] - An array of broadcastEmailIds to be deleted.
+* `callback`: [function] - callback to be executed when the operation completes.
+
+### Examples
+```javascript
+// ~~~~~
+// Deletes a broadcastEmail item.
```
Comment (matthopson) [inline docs/collections/broadcast_emails.md]: Should be something like
```javascript
// Deletes multiple broadcastEmail items.
```

File: docs/collections/divisions.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+---
+
+
+<a id="ancestorDivisions"></a>
+## `ancestorDivisions(divisionId, callback)`
+Returns the ancestor `division` items.
+
+### Params
+* `divisionId`: [int] - an divisionId.
```
Comment (matthopson) [inline docs/collections/divisions.md]: "a" divisionId

File: docs/collections/divisions.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+---
+
+
+<a id="descendantDivisions"></a>
+## `descendantDivisions(divisionId, callback)`
+Returns the descendant `division` items.
+
+### Params
+* `divisionId`: [int] - an divisionId.
```
Comment (matthopson) [inline docs/collections/divisions.md]: "a" divisionId

File: docs/collections/divisions.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+* `callback`: [function] - callback to be executed when the operation completes.
+### Examples
+```javascript
+// ~~~~~
+// Returns the ancestor divisions.
```
Comment (matthopson) [inline docs/collections/divisions.md]: "descendant" divisions

File: docs/collections/divisions.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+---
+
+
+<a id="childrenDivisions"></a>
+## `childrenDivisions(divisionId, callback)`
+Returns the child `division` items.
+
+### Params
+* `divisionId`: [int] - an divisionId.
```
Comment (matthopson) [inline docs/collections/divisions.md]: "a"

File: docs/collections/divisions.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+* `callback`: [function] - callback to be executed when the operation completes.
+### Examples
+```javascript
+// ~~~~~
+// Returns the ancestor divisions.
```
Comment (matthopson) [inline docs/collections/divisions.md]: "child" divisions

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,14 @@
 # TeamSnap JavaScript SDK CHANGELOG
 
+### September 30, 2016 // Version 1.17.0
+- Add `divisions` endpoint.
+- Update validations on `broadcastEmail`.`saveBroadcastEmail`.
+- Add `bulkDeleteBroadcastEmails` method.
+- Add `divisionSearchMembers` method.
+- Add `divisionSearchTeams` method.
+
```
Comment (matthopson) [inline CHANGELOG.md]: No mention of children, ancestor, or descendent methods.

File: src/collections/divisions.coffee [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+exports.descendantDivisions = (divisionId, callback) ->
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
+  @collections.divisions.queryItems('descendants', id: divisionId, callback)
+
+
+exports.childrenDivisions = (divisionId, callback) ->
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
```
Comment (matthopson) [inline src/collections/divisions.coffee]: Should be `teamsnap.childrenDivisions`

File: src/collections/divisions.coffee [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+exports.ancestorDivisions = (divisionId, callback) ->
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
+  @collections.divisions.queryItems('ancestors', id: divisionId, callback)
+
+
+exports.descendantDivisions = (divisionId, callback) ->
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
```
Comment (matthopson) [inline src/collections/divisions.coffee]: Should be `teamsnap.descendantDivisions`

File: src/collections/divisions.coffee [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  unless division
+    throw new TSArgsError 'teamsnap.deleteDivision',
+      '`division` must be provided'
+
+  @deleteItem division, callback
+
+
+exports.ancestorDivisions = (divisionId, callback) ->
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
```
Comment (matthopson) [inline src/collections/divisions.coffee]: Should be `teamsnap.ancestorDivisions`

File: src/collections/divisions.coffee [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+exports.deleteDivision = (division, callback) ->
+  unless division
+    throw new TSArgsError 'teamsnap.deleteDivision',
+      '`division` must be provided'
+
+  @deleteItem division, callback
+
+
+exports.ancestorDivisions = (divisionId, callback) ->
```
Comment (matthopson) [inline src/collections/divisions.coffee]: I feel like this method should be called `loadAncestorDivisions` to maintain consistency with outer load methods.

File: src/collections/divisions.coffee [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
+  @collections.divisions.queryItems('ancestors', id: divisionId, callback)
+
+
+exports.descendantDivisions = (divisionId, callback) ->
```
Comment (matthopson) [inline src/collections/divisions.coffee]: Same as ☝️ - `loadDescendantDivisions`

File: src/collections/divisions.coffee [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+  unless @isId divisionId
+    throw new TSArgsError 'teamsnap.loadDivision', 'divisionId must be provided'
+  @collections.divisions.queryItems('descendants', id: divisionId, callback)
+
+
+exports.childrenDivisions = (divisionId, callback) ->
```
Comment (matthopson) [inline src/collections/divisions.coffee]: Same as ☝️ - loadChildrenDivisions

File: src/collections/members.coffee [hunk 15]
```diff
@@ -97,6 +97,13 @@ exports.disableMember = (memberId, callback) ->
   @collections.members.exec('disableMember', params)
     .pop().callback callback
 
+exports.divisionSearchMembers = (params, callback) ->
```
Comment (nathanielvelez) [inline src/collections/members.coffee]: wouldn't it read better if this method was called `searchDivisionMembers`?

File: src/collections/teams.coffee [hunk 16]
```diff
@@ -125,6 +126,15 @@ exports.resetStatistics = (teamId, callback) ->
   @collections.teams.exec('resetStatistics', params)
   .callback callback
 
+
+exports.divisionSearchTeams = (params, callback) ->
```
Comment (nathanielvelez) [inline src/collections/teams.coffee]: wouldn't it read better if this method was called `searchDivisionTeams`?

File: src/collections/members.coffee (hunk 15 above)
Comment (matthopson) [inline src/collections/members.coffee]: I wonder if we're using "search" in place of "load" a lot

Comment (matthopson) [inline src/collections/members.coffee]: I agree - or possibly `loadDivisionMembers`

File: src/collections/teams.coffee (hunk 16 above)
Comment (matthopson) [inline src/collections/teams.coffee]: Same opinion as ☝️ - would like to consider `loadDivisionTeams`

File: src/collections/members.coffee (hunk 15 above)
Comment (dmccraw) [inline src/collections/members.coffee]: I was following the convention of the action followed by the endpoint `searchDivision` on `members` endpoint. The only problem I see with searchDivisionMembers is that we don't really have a divisionMember just a member that happens to be in a division.

Comment (nathanielvelez) [inline src/collections/members.coffee]: 👍

Comment (matthopson) [inline src/collections/members.coffee]: @dmccraw I thought that might be where you were going with that. I'm cool with it. What are your thoughts on the "load / search" thing?

**Comment by dmccraw** [comment]
The more I thought about it the more I liked the names `searchDivisionTeams` and `searchDivisionMembers`. @nathanielvelez updated

**Comment by matthopson** [comment]
@nathanielvelez You completely ignored all the other comments.

**Comment by dmccraw** [comment]
Which open comments?

**Comment by matthopson** [comment]
@dmccraw  are you not seeing these?
screen shot 2016-10-04 at 11 40 39 am

**Comment by dmccraw** [comment]
Nope, I didn't see any of those and didn't get any emails regarding them.
Did you finish your review?

**Comment by matthopson** [comment]
@nathanielvelez @dmccraw  Dang - that's my bad. I didn't realize it wasn't posting them. First time using the "start review" comment.

File: src/collections/members.coffee (hunk 15 above)
Comment (dmccraw) [inline src/collections/members.coffee]: Ok, I see how `load` is always used instead of `search` in the sdk.

**Pull request merged**